src/main/java/com/openlattice/edm/type/PropertyType.java

```diff
@@ -89,7 +83,11 @@ public PropertyType(
         this.piiField = piiField.orElse( false );
         this.multiValued = multiValued.orElse( true );
         this.analyzer = analyzer.orElse( Analyzer.STANDARD );
-        this.postgresIndexType = postgresIndexType.orElse( IndexType.BTREE );
+        if ( EdmPrimitiveTypeKind.Binary == this.datatype ) {
```

Comment:
I might do this as a checkArgument or switch to .orElseGet( ... ) that can take a lambda with the if statement. The only issue with the current approach is that it silently sets the datatype instead of throwing the error.

Reply:
Yep, that makes more sense for sure.